src/caseManagement.js

```diff
@@ -0,0 +1,510 @@
+            });
+            form.on('question-create', function (e) {
+                // this will get called when a deletion is undone.
+                // Ensure that we restore the previously deleted mappings, if present
+                const maintainer = new CaseMapMaintainer(form);
+                const mug = e.mug;
+                const case_property = mug.p.case_property;
+                if (case_property) {
+                    maintainer.updateFormMappings(mug.absolutePath, null, case_property);
+                }
+            });
+        },
+
+        performAdditionalParsing: function (form, xml) {
+            this.__callOld();
+
+            if (!this.data.caseManagement.isActive) {
+                return;
+            }
+
+            const builder = new CaseMappingsBuilder();
+            builder.addMappingsToForm(form, xml);
+        },
+
+        contributeToAdditionalXML: function (xmlWriter, form) {
+            this.__callOld();
+
+            if (!this.data.caseManagement.isActive) {
+                return;
+            }
+
+            const writer = new XMLCaseMappingWriter(xmlWriter);
+            writer.writeCaseMappingsElement(form);
+        },
+
+        getMugTypes: function () {
+            const types = this.__callOld();
+            // never show the case management property for labels
+            types.normal.Trigger.spec.case_property = { presence: 'notallowed' };
+
+            const groupTypes = [
+                types.normal.Group,
+                types.normal.Repeat,
+                types.normal.FieldList
+            ];
+            groupTypes.forEach(groupType => groupType.spec.case_property = { presence: 'notallowed' });
+
+            return types;
+        },
+
+        getSections: function (mug) {
+            const sections = this.__callOld(mug);
+
+            if (!this.data.caseManagement.isActive) {
+                return sections;
+            }
+
+            sections.splice(1, 0, {
+                slug: 'caseManagement',
+                displayName: gettext('Case Management'),
+                properties: ['case_property'],
+                help: {
+                    title: gettext('Case Management'),
+                    text: gettext(
+                        'Saving a question as a case property makes it accessible and reusable ' +
+                        'across your application. To manage all your case properties from this form, ' +
+                        'click on the Manage Case button. '
+                    ),
+                    link: 'https://dimagi.atlassian.net/wiki/spaces/commcarepublic' +
+                        '/pages/2143955170/Case+Management+Overview',
+                }
+            });
+
+            return sections;
+        },
+
+        getMugSpec: function () {
+            const specs = this.__callOld();
+            const that = this;
+
+            const databindSpecs = Object.assign(specs.databind, {
+                'case_property': {
```

Comment:
nit: Most mug properties use `camelCase`, although a few inconsistent ones use `snake_case`. Why did you decide to go with `case_property`?

Reply:
Probably an oversight, or just muscle memory from the python side. Changed to camelCase: abe700051aba65f22b7aa46909ba6feff2aefbee